examples/lock-app/k32w/main/main.cpp

```diff
@@ -65,6 +65,8 @@ extern "C" void main_task(void const * argument)
         (*pFunc)();
     }
 
+    mbedtls_platform_set_calloc_free(otPlatCAlloc, otPlatFree);
```

Comment:
Why go through otPlatCalloc/otPlatFree? The mbedtls library should use whatever this platform's allocator is, and NOT rely on implementations from OpenThread, which, as it stands, are no longer supported to be called outside of OpenThread IIUC.

```suggestion
    mbedtls_platform_set_calloc_free(pvPortCallocRtos, vPortFree);
```

Reply:
Shouldn't we use CHIPPlatformMemoryCalloc and CHIPPlatformMemoryFree as defined in src/lib/support/CHIPPlatformMemory.cpp

You can also map otPlatCAlloc and otPlatFree to use those to so it is consistent a cross the project

Reply:
Change the call to mbedtls_platform_set_calloc_free(pvPortCallocRtos, vPortFree) as suggested by @tcarmelveilleux.

If I use CHIPPlatformMemoryCalloc then a call to malloc is done internally and I'm not sure if this is redirected to FreeRTOS pvPortMalloc.

Reply:
Below is the pvPortMalloc function from heap3.c (FreeRTOS). In addition to calling malloc is also do some task related processing. So I believe that it's not safe to call directly CHIPPlatformMemoryCalloc instead of pvPortMalloc.
void *pvPortMalloc( size_t xWantedSize )
{
void *pvReturn;

	vTaskSuspendAll();
	{
		pvReturn = malloc( xWantedSize );
		traceMALLOC( pvReturn, xWantedSize );
	}
	( void ) xTaskResumeAll();

	#if( configUSE_MALLOC_FAILED_HOOK == 1 )
	{
		if( pvReturn == NULL )
		{
			extern void vApplicationMallocFailedHook( void );
			vApplicationMallocFailedHook();
		}
	}
	#endif

	return pvReturn;
}

Reply:
There are a few points here:

- `CHIPPlatformMemoryCalloc` calls `chip::platform::MemoryCalloc()` which calls `calloc()`.
- pvPortMalloc on FreeRTOS, needs to be tied to replace the C/C++ library malloc, so that you don't have 2 heaps: FreeRTOS and C library "leftover linker space" in the linker script as an untracked heap. The other reason it needs to be replaced is that pvPortMalloc does the necessary processing to make malloc concurrency-safe on FreeRTOS.

It is odd to me that these allocators (e.g. `chip::platform::MemoryCalloc()`) are using global functions scoping and end-up calling C library, which is what you are trying to defeat by directly calling pvPortCallocRtos/vPortFree in other places.